activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -147,6 +151,8 @@ def initialize(connection)
       end
 
       def begin_transaction(options = {})
+        options[:run_callbacks] ||= !current_transaction.joinable?
```

Comment:
this will allow anyone that creates a new transaction to force that transaction to run callbacks. I think this is a good feature to have. But I cannot think about use cases, so I am not sure if we should add it or not.. In any case, if we dont want that feature I can just remove the `||` from this line.